### DISCUSSION TITLE
Nix build from master fails due to invalid HEAD revision for gabmus/tree-sitter-blueprint

### DISCUSSION THREAD

**Discussion by koalazub** [helix-editor/helix#11554] [Troubleshooting] [answered]
### Summary

The latest HEAD revision being referenced here is incorrect:
https://github.com/helix-editor/helix/blob/38e6fcd5c51478635ffa405815c7b9bbeadc35a9/languages.toml#L3137

And at the time of writing is: 

`60ba73739c6083c693d86a1a7cf039c07eb4ed59` from : https://gitlab.com/gabmus/tree-sitter-blueprint

### Reproduction Steps

Pull the flake from master helix-editor/helix repo via flake.nix: 

in flake.nix: 

```nix
{
  description = "system configuration flake";

  inputs = {
    helix.url = "github:helix-editor/helix";
    ...
  };

  outputs =
    { self, helix, ... }:
    {
      ...
    };
}
```

`nix develop` and `nix build` both don't work

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

macOS Version 15.0 Beta (24A5309e)

### Terminal Emulator

alacritty 0.13.2

### Installation Method

nix-darwin flake 

### Helix Version

helix 24.7 (079f5442)

**Comment by the-mikedavis** [comment]
What version of Nix and git do you have? I was able to build from scratch with no cache using whatever the current latest Nix container is.

That `rev` shouldn't necessarily point to HEAD - it's meant to lock the repository at any version in the cmomit history. The listed revision still exists in the log: https://gitlab.com/gabmus/tree-sitter-blueprint/-/commit/863cea9f83ad5637300478e0559262f1e791684b

**Reply by koalazub** [reply]
git version 2.39.5 (Apple Git-154)
nix (Nix) 2.18.5
darwin version: 24.11.20240818.8a33541+darwin4.d6703b9

I forgot to add the error log I was receiving too: 

```
ssh: Could not resolve hostname gitlab.comgabmus: nodename nor servname provided, or not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
warning: could not update mtime for file '/var/root/.cache/nix/gitv3/05ihl1gim35rjihrp59gnaygf3xhbbjwkr118p0gg5c21y4sw49c-shallow/refs/heads/HEAD': No such file or directory

...

       error: Cannot find Git revision '863cea9f83ad5637300478e0559262f1e791684b' in ref 'HEAD' of repository 'https://gitlab.com/gabmus/tree-sitter-blueprint'! Please make sure that the rev exists on the ref you've specified or add allRefs = true; to fetchGit.

```

When I made the mention of this in the NixOS chat they mentioned something about the author of the package force pushing and to opt for waiting it out. But now it's been a week and I can't build from master. 

I've already cleared my nix cache via `nix-collect-garbage -d` and attempted to build other repos and that was fine. It's just with helix that I'm finding to be problematic. 

Running `nix flake show` runs just fine: 

```
nix flake show
git+file:///Users/myUser/projects/helix?ref=refs/heads/master&rev=38e6fcd5c51478635ffa405815c7b9bbeadc35a9
├───checks
│   ├───aarch64-darwin
│   │   ├───clippy: derivation 'helix-term-clippy-24.7.0'
│   │   ├───doc: derivation 'helix-term-doc-24.7.0'
│   │   ├───fmt: derivation 'helix-term-fmt-24.7.0'
│   │   ├───helix: derivation 'helix-term-24.7.0'
│   │   └───test: derivation 'helix-term-test-24.7.0'
```

There's too much going on here. So I guess your statement makes me wonder if this is an issue with my machine or if there's an issue with building the flake. That's what I want to home in on. 

Just for the sake of sanity, I'll go through reinstalling nix/nix-darwin and report back. But the fact that it's only this flake vs my other dev flakes in other repos building is concerning me here.

**Reply by koalazub** [reply]
Also just to save you the splatter of text here's the summary: 

1. Attempted to build Helix from master, encountering an error.

2. Reported the issue in the NixOS chat, where it was suggested that the package author may have force pushed.

3. Waited for a week after the initial report, but the issue persisted.

4. Cleared your Nix cache using `nix-collect-garbage -d`.

5. Attempted to build other repositories, which were successful.

6. Ran `nix flake show`, which executed without errors.

7. Considered reinstalling nix/nix-darwin as a potential solution.

8. Noted that the issue appears to be specific to the Helix flake, as other development flakes in other repositories are building successfully.

**Comment by koalazub** [comment]
I'm still getting the same error message after fully reinstalling nix-darwin: 

```
nix (Nix) 2.18.5

       error: Cannot find Git revision '863cea9f83ad5637300478e0559262f1e791684b' in ref 'HEAD' of repository 'https://gitlab.com/gabmus/tree-sitter-blueprint'! Please make sure that the rev exists on the ref you've specified or add allRefs = true; to fetchGit.
```
Have you got thoughts on anything that could affect this externally? I've commented above on how I have the flake setup to show that I'm not pinning anything and just letting the flake do the work

**Reply by the-mikedavis** [reply]
My only other thought is that it might be a bug in Nix itself. I see some fetchTree / fetchGit issues in the NixOS/nix tracker but I'm not sure they affect the Nix versions we're using. An easy way to (hopefully) fix this would be to bump the revision for tree-sitter-blueprint here: https://github.com/helix-editor/helix/blob/1b5295a3f3d7cccd96eed5bfd394807a4dae87fc/languages.toml#L3137

to the current latest revision `60ba73739c6083c693d86a1a7cf039c07eb4ed59`. There's only one commit between the current revision we're locked on and it's a fix to a regex in `grammar.js` so it doesn't need query changes. Could you give that a try locally and see if that fixes the fetchGit failure?

**Comment by koalazub** [comment, marked as answer]
Turns out the issue was all my doing. 

The issue was related to git config: 

```bash
[url "ssh://git@gitlab.com/"]
    insteadOf = https://gitlab.com/
```

changed to: 

```bash
[url "ssh://git@gitlab.com/"]
    pushInsteadOf = https://gitlab.com/
```

Which was causing the conflicts when building the project via `nix` and pulling each language via git. Where it was respecting the `gitconfig` settings to pull via ssh without a fallback. Causing the error: 

```nushell
ssh: Could not resolve hostname gitlab.comgabmus: nodename nor servname provided, or not known
fatal: Could not read from remote repository.
```

I just tested on the 'problematic' commit hash and it's running fine as well as the newest commit version.
The thing that threw me off was what looks like a typo? `gitlab.comgabmus` or maybe a problematic output. But in the end was just the config
